docs/data-types.md

```diff
@@ -0,0 +1,332 @@
+## Basic data types
```

Comment:
Begin with `# Datatypes`, so nav item is also named as `Datatypes`

Reply:
Done.